core/trino-spi/src/main/java/io/trino/spi/type/ShortTimestampType.java

```diff
@@ -125,6 +129,12 @@ public Object getObjectValue(ConnectorSession session, Block block, int position
         return SqlTimestamp.newInstance(getPrecision(), epochMicros, 0);
     }
 
+    @Override
+    public Optional<Stream<?>> getDiscreteValues(Range range)
+    {
+        return Optional.of(LongStream.rangeClosed((long) range.getMin(), (long) range.getMax()).boxed());
```

Comment:
That's correct for `timestamp(6)` only

Reply:
Thanks for adding a check. If `getDiscreteValues` is valuable for timestamps, add a TODO note and file an issue to implement it correctly for other precisions (incl long timestamp).
(If it's not valuable, you can remove it completely :) )

Reply:
We didn't use this feature (expanding/merging of consecutive values) for time/timestamp columns, so I prefer to keep it unimplemented for now.